java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -68,20 +68,33 @@
 
 public class SymbolicEvaluator {
 
+  private static final int MAXIMAL_EXECUTION_STATE_COUNT = 65536;
+
   private final AssignedSymbolExtractor extractor = new AssignedSymbolExtractor();
 
   private final Map<Tree, SymbolicBooleanConstraint> result = new HashMap<>();
 
+  private int currentExecutionStateCount;
+  private int maximalExecutionStateCount;
+
+  public SymbolicEvaluator() {
+    maximalExecutionStateCount = MAXIMAL_EXECUTION_STATE_COUNT;
+  }
+
   public Map<Tree, SymbolicBooleanConstraint> evaluateMethod(ExecutionState state, MethodTree tree) {
     result.clear();
     if (tree.block() != null) {
-      evaluateStatement(ImmutableList.of(state), tree.block());
+      try {
+        evaluateStatement(ImmutableList.of(state), tree.block());
+      } catch (SymbolicExecutionException e) {
```

Comment:
We cannot, since the exception should be propagated through the visit methods.

Anyway the exception is not supposed to leak outside the visitor.